### PR TITLE
Fixes #24247: Authentication protocol names should be normalized to lowercase

### PR DESCRIPTION
**Pull request by clarktsiory**
https://issues.rudder.io/issues/24247

### CODE REVIEW

**Comment by clarktsiory** [comment]
PR updated with a new commit

File: auth-backends/src/main/scala/bootstrap/rudder/plugin/AuthBackendsConf.scala [hunk 1]
```diff
@@ -616,7 +616,7 @@ class RudderOidcUserService(
   // we need to use our copy of DefaultOAuth2UserService to log/manage errors
   super.setOauth2UserService(new RudderDefaultOAuth2UserService())
 
-  override val protocolName = "OIDC"
+  override val protocolName = "oidc"
```
Comment (fanf) [inline auth-backends/src/main/scala/bootstrap/rudder/plugin/AuthBackendsConf.scala:619]: This is a human readable name used in debug/log, so this one should be as it is in the litterature. 
The problem is likely that we should not use that for the thing we put in the database.

Comment (fanf) [reply auth-backends/src/main/scala/bootstrap/rudder/plugin/AuthBackendsConf.scala:619]: I think what we want is the key of the authentication provider stored in `RudderConfig.authenticationProviders` used for the authentication

**Review by fanf — APPROVED** [review]
Neat!

**Review by fanf — APPROVED** [review]

**Comment by fanf** [comment]
Merging by hand because feature branch

**Pull request merged**